docs/en/04_Changelogs/4.1.0.md

```diff
@@ -0,0 +1,87 @@
+# 4.1.0
+
+## Introduction
+
+This version introduces many breaking changes, which in most projects can be managed through a combination
+of automatic upgrade processes as well as manual code review. This document reviews these changes and will
+guide developers in preparing existing 3.x code for compatibility with 4.0
+
+## Overview {#overview}
+
+ * Support for public webroot folder `public/`
+ * Better support for cross-platform filesystem path manipulation
+
+## Upgrading
+
+### Upgrade `public/` folder
+
+By default all new projects installed from 4.1.0 will have a `public/` webroot folder. This folder name is
+not configurable, but is turned on by creating this file, and off by ensuring this folder doesn't exist.
+
+When separating the public webroot from the BASE_PATH it is necessary to move a few files during migration:
+
+ - Move `.htaccess` from base to `public/`
+ - Move `index.php` from base to `public/`
+ - Move `assets` folder (including the nested `assets/.protected` folder) into `public/`
+ - Ensure that the `public/resources` folder exists; If this folder already exists in root, you should
+   delete this, and re-generate it by running `composer vendor-expose` in your root path.
+ - Any public assets committed directly to your project intended to be served directly to the
+ webserver. E.g. move `mysite/javascript/script.js` to `public/javascript/script.js`.
+ - Ensure that the web-root configured for your server of choice points to the public/ folder instead of the base path.
+ 
+You may also need to add various changes to your code if you reference the BASE_PATH directly:
+
+ - You should use `Director::publicFolder()` instead of `Director::baseFolder()` if referring to the
+ public folder.
+ - You can check if a public folder exists with `Director::publicDir()`
+ 
+### Use new `$public` theme set
+ 
+In addition there is a new helper pseudo-theme that you can configure to expose files in the `public/`
+folder to the themed css / javascript file lookup. For instance, this is how you can prioritise those
+files:
+
+```yaml
+---
+Name: mytheme
+---
+SilverStripe\View\SSViewer:
+  themes:
+    - '$public'
+    - 'simple'
+    - '$default'
+```
+
+This would allow `<% require themedCSS('style.css') %>` to find a file comitted to `public/css/style.css`.
+
+Note that `Requirements` calls will look in both the `public` folder (first) and then the base path when
+resolving css or javascript files.
+
+### Expose root project files
```

Comment:
* Describe how to detect modules which might state compatibility with 4.x, but are using public files without using `expose` yet. Also make it really clear that module authors need to switch to `silverstripe-vendormodule` now, particularly if they're exposing public files.